.circleci/config.yml

```diff
@@ -653,165 +660,20 @@ jobs:
-            git --no-pager diff --name-only HEAD $(git merge-base HEAD origin/master) | sort > /tmp/changed_files
-            echo ".circleci/config.yml" >> /tmp/ci_docker_files
-            echo "cargo-toolchain" >> /tmp/ci_docker_files
-            echo "docker/ci/alpine/Dockerfile" >> /tmp/ci_docker_files
-            echo "docker/ci/arch/Dockerfile" >> /tmp/ci_docker_files
-            echo "docker/ci/centos/Dockerfile" >> /tmp/ci_docker_files
-            echo "docker/ci/Dockerfile" >> /tmp/ci_docker_files
-            echo "rust-toolchain" >> /tmp/ci_docker_files
-            echo "scripts/dev_setup.sh" >> /tmp/ci_docker_files
-
-            #if no relevant files have changed halt this build and mark it success, hopefully effects dependent jobs
-            if [ `join /tmp/changed_files /tmp/ci_docker_files | wc -l` == 0 ]; then
-              echo no relevant files have changed - exit
-              circleci step halt
-            fi
-
-  ######################################################################################################
-  # Publish docker ci base image to docker hub                                                         #
-  ######################################################################################################
-  docker-ci-publish:
-    executor: docker-executor
-    description: publish docker images
-    steps:
-      - checkout
-      - setup_docker_signing
-      - run:
-          name: Build the ci base image to prevent errors in the nightly build.
-          command: |
-
-            #detect if any relevant files have changed requiring base docker file rebuilding.
-            #this is specific for prs targeting master.   TODO: figure out target release branch.
-
-            git --no-pager diff --name-only HEAD $(git merge-base HEAD origin/master) | sort > /tmp/changed_files
-            echo ".circleci/config.yml" >> /tmp/ci_docker_files
-            echo "cargo-toolchain" >> /tmp/ci_docker_files
-            echo "docker/ci/Dockerfile" >> /tmp/ci_docker_files
-            echo "rust-toolchain" >> /tmp/ci_docker_files
-            echo "scripts/dev_setup.sh" >> /tmp/ci_docker_files
-
-            #if no relevant files have changed halt this build and mark it success, hopefully effects dependent jobs
-            if [ `join /tmp/changed_files /tmp/ci_docker_files | wc -l` == 0 ]; then
-              echo no relevant files have changed - exit
-              circleci step halt
-            fi
-            set -x
-            #building with content trust tends to break in circleci
-            export DOCKER_CONTENT_TRUST=0
-            docker build -f docker/ci/Dockerfile -t libra/build_environment:latest .
             #signs the pushed image
-            docker push --disable-content-trust=false libra/build_environment:latest
+            docker push --disable-content-trust=false libra/build_environment:<< parameters.base-image >>-latest
```

Comment:
Need to think about DockerHub image retention policy soon :)

Reply:
I'm aware.   This only builds on changes to the relevant files.

Reply:
I'm not done with the nightly run yet that will be triggered on master to handle the cleanups.   It's a bit of a pia, and got side tracked with lcov stuff today.